src/Playwright/Core/Locator.cs

```diff
@@ -51,22 +54,23 @@ public Locator(Frame parent, string selector, LocatorLocatorOptions options = nu
 
         if (options?.HasTextRegex != null)
         {
-            var jsRegex = $"/{options.HasTextRegex.ToString()}/{options.HasTextRegex.Options.GetInlineFlags()}";
-            _selector += $" >> has={JsonSerializer.Serialize("text=" + jsRegex, serializerOptions)}";
+            var textSelector = "text=" + EscapeForTextSelector(options.HasTextRegex, false);
```

Comment:
nit: You don't need the branching anymore since EscapeForTextSelector takes care of it